rules-tests/TypeDeclaration/Rector/FunctionLike/AddClosureParamTypeForArrayMapRector/Fixture/fixture.php.inc

```diff
@@ -0,0 +1,87 @@
+<?php
+
+namespace Rector\Tests\TypeDeclaration\Rector\FunctionLike\AddClosureParamTypeForArrayMapRector\Fixture;
+
+use Rector\Tests\TypeDeclaration\Rector\FunctionLike\AddClosureParamTypeForArrayMapRector\Source\Bar;
+use Rector\Tests\TypeDeclaration\Rector\FunctionLike\AddClosureParamTypeForArrayMapRector\Source\Foo;
+
+class Fixture
+{
+    /**
+     * @param array<int, string> $array
+     */
+    public function run(array $array)
+    {
+        return array_map(function ($value, $key) {
+            return $value . $key;
+        }, $array);
+    }
+
+    /**
+     * @param array<int, string> $array
+     * @param array<int, Foo> $arrayTwo
+     */
+    public function runTwo(array $array, array $arrayTwo)
+    {
+        return array_map(function ($value, $key) {
+            return get_class($value) . $key;
+        }, $array, $arrayTwo);
+    }
+
+    /**
+     * @param array<int, string> $array
+     * @param array<int, Foo|Bar> $arrayTwo
+     */
+    public function runThree(array $array, array $arrayTwo)
+    {
+        return array_map(function ($value, $key) {
+            return get_class($value) . $key;
+        }, $array, $arrayTwo);
+    }
+}
+
+?>
+-----
+<?php
+
+namespace Rector\Tests\TypeDeclaration\Rector\FunctionLike\AddClosureParamTypeForArrayMapRector\Fixture;
+
+use Rector\Tests\TypeDeclaration\Rector\FunctionLike\AddClosureParamTypeForArrayMapRector\Source\Bar;
+use Rector\Tests\TypeDeclaration\Rector\FunctionLike\AddClosureParamTypeForArrayMapRector\Source\Foo;
+
+class Fixture
+{
+    /**
+     * @param array<int, string> $array
+     */
+    public function run(array $array)
+    {
+        return array_map(function (string $value, int $key) {
+            return $value . $key;
+        }, $array);
+    }
+
+    /**
+     * @param array<int, string> $array
+     * @param array<int, Foo> $arrayTwo
+     */
+    public function runTwo(array $array, array $arrayTwo)
```

Comment:
Could you add a test fixture for following:

```php
* @param array<int> $array
```

I think these should be skipped, right?

Reply:
I've added these to fixtures. It comes about it two different ways.

```php
/**
     * @param array<int, string> $array
     * @param array<string> $arrayTwo // translates to the key is either int or string and the value is string
     * @param list<string> $arrayTwo // translates to the key is either int and the value is string
     */
```

Reply:
I've added another fixture for tuples and array spread. I'd love to make that work but it's just too complicated. I've had a dig into PHPStan and how it gets the types but it's not overly straight forward to understand how its own classes work out the types.